src/OpenCensus.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -0,0 +1,161 @@
+            span.PutHttpUrlAttribute(request.RequestUri.ToString());
+        }
+
+        public override void OnStopActivity(Activity activity, object payload)
+        {
+            if (!(this.stopResponseFetcher.Fetch(payload) is HttpResponseMessage response))
+            {
+                // Debug.WriteLine("response is null");
+                return;
+            }
+
+            var requestTaskStatus = this.stopRequestStatusFetcher.Fetch(payload) as TaskStatus?;
+
+            var span = this.Tracer.CurrentSpan;
+            if (requestTaskStatus.HasValue)
+            {
+                if (requestTaskStatus != TaskStatus.RanToCompletion)
+                {
+                    span.PutErrorAttribute(requestTaskStatus.ToString());
+                }
+            }
+
+            if ((int)response.StatusCode >= 200 && (int)response.StatusCode <= 301)
+            {
+                span.Status = Status.Ok;
+            }
+            else if ((int)response.StatusCode == 401)
+            {
+                span.Status = Status.Unauthenticated;
+            }
+            else if ((int)response.StatusCode == 403)
+            {
+                span.Status = Status.PermissionDenied;
+            }
+            else if ((int)response.StatusCode == 404)
+            {
+                span.Status = Status.NotFound;
+            }
+            else if ((int)response.StatusCode == 501)
+            {
+                span.Status = Status.Unimplemented;
+            }
+            else
+            {
+                span.Status = Status.Unknown;
+            }
+
+            span.Status = span.Status.WithDescription(response.StatusCode + " " + response.ReasonPhrase);
+
+            span.PutHttpStatusCodeAttribute((int)response.StatusCode);
+            this.LocalScope.Value?.Dispose();
+        }
+
+        public override void OnStopActivityWithException(Activity activity, object payload)
+        {
+            if (!(this.stopExceptionFetcher.Fetch(payload) is Exception exc))
+            {
+                // Debug.WriteLine("response is null");
+                return;
+            }
+
+            var span = this.Tracer.CurrentSpan;
+            var requestTaskStatus = this.stopRequestStatusFetcher.Fetch(payload) as TaskStatus?;
+            if (requestTaskStatus.HasValue)
+            {
+                if (requestTaskStatus != TaskStatus.RanToCompletion)
+                {
+                    span.PutErrorAttribute(requestTaskStatus.ToString());
+                }
+            }
+
+            this.LocalScope.Value?.Dispose();
+
+            if (exc is HttpRequestException)
+            {
+                // this will be System.Net.Http.WinHttpException: The server name or address could not be resolved
+                // on netstandard...
+                if (exc.InnerException is WebException &&
+                    ((WebException)exc.InnerException).Status == WebExceptionStatus.NameResolutionFailure)
```

Comment:
would it work on Core? It did not work 1,5 years ago. Also it'd be great to have WebException status in the span status.

Reply:
nope. I put a comment for this

Reply:
so perhaps this would never happen on this DiagnosticListener - this is core only